daemon/emer-daemon.c

```diff
@@ -1123,6 +1142,379 @@ set_max_bytes_buffered (EmerDaemon *self,
+  while (g_hash_table_iter_next (&iter, NULL, (gpointer *) &timer_data))
+    {
+      save_aggregate_timer_to_tally (timer_data);
+      timer_data->start_monotonic_us = g_get_monotonic_time ();
+    }
+}
+
+static void
+buffer_daily_aggregate_events_to_queue (EmerDaemon *self)
+{
+  EmerDaemonPrivate *priv = emer_daemon_get_instance_private (self);
+  g_autoptr(GFileEnumerator) enumerator = NULL;
+  g_autoptr(GError) error = NULL;
+  g_autoptr(GFile) tallies_folder = NULL;
+  g_autofree gchar *tallies_path = NULL;
+  g_autofree gchar *date_string = NULL;
+
+  date_string = g_date_time_format (priv->current_aggregate_tally_date,
+                                    "%Y-%m-%d");
+  tallies_path = g_build_filename (priv->persistent_cache_directory,
+                                   "aggregate-timers",
+                                   date_string,
+                                   NULL);
+
+  tallies_folder = g_file_new_for_path (tallies_path);
+
+  enumerator = g_file_enumerate_children (tallies_folder,
+                                          "standard::*",
+                                          G_FILE_QUERY_INFO_NONE,
+                                          NULL,
+                                          &error);
+
+  if (error)
+    {
+      if (!g_error_matches (error, G_IO_ERROR, G_IO_ERROR_NOT_FOUND))
+        g_warning ("Error listing tallies from %s: %s",
+                   tallies_path, error->message);
+      return;
+    }
+
+  while (TRUE)
+    {
+      g_autoptr(GVariant) variant = NULL;
+      g_autoptr(GVariant) cache_key = NULL;
+      g_autoptr(GVariant) event_id = NULL;
+      g_autoptr(GVariant) aggregate_key = NULL;
+      g_autoptr(GVariant) payload = NULL;
+      g_autoptr(GError) local_error = NULL;
+      g_autofree gchar *contents = NULL;
+      g_autofree gchar *path = NULL;
+      GFileInfo *info;
+      GFile *file;
+      gboolean complete;
+      guint32 counter;
+      uid_t unix_user_id;
+
+      g_file_enumerator_iterate (enumerator, &info, &file, NULL, &local_error);
+
+      if (local_error)
+        {
+          if (!g_error_matches (error, G_IO_ERROR, G_IO_ERROR_NOT_FOUND))
+            g_warning ("Error listing tallies from %s: %s",
+                       tallies_path, local_error->message);
+          return;
+        }
+
+      if (!file || !info)
+        break;
+
+      if (g_file_info_get_file_type (info) != G_FILE_TYPE_REGULAR)
+        continue;
+
+      path = g_file_get_path (file);
+      g_file_get_contents (path, &contents, NULL, &local_error);
+
+      if (local_error)
+        {
+          g_warning ("Error loading tally from file %s: %s",
+                     path, local_error->message);
+          continue;
+        }
+
+      variant = g_variant_parse (NULL, contents, NULL, NULL, &local_error);
+      if (local_error)
+        {
+          g_warning ("Error parsing tally from file %s: %s",
+                     path, local_error->message);
+          continue;
+        }
```

Comment:
At some point we should presumably try to delete broken files in this directory to stop them accumulating.